Scenarios/BlueGreen-Deployment-for-AKS/GAP/bluegreen-aks-solution-content.md

```diff
@@ -0,0 +1,222 @@
+## Overview 
+What happen when you need to upgrade the kubernetes version in a cluster or even better when you need to change kubernetes platform components like: ingress gateway, service mesh, operators and so on, but you don't want to have impact on the workloads/applications that are running on the cluster itself?
+
+The answer that will make everyone happy from business into infra to apps is blue green deployment at Kubernetes infra level.
+With modern principles and availability of cloud services like:
+- [IaC](https://docs.microsoft.com/en-us/devops/deliver/what-is-infrastructure-as-code)
+- [Immutable Infrastructure](https://www.hashicorp.com/resources/what-is-mutable-vs-immutable-infrastructure)
+- [Cloud Elasticity](https://azure.microsoft.com/en-us/overview/what-is-elastic-computing/)
+- [Continuous Delivery](https://docs.microsoft.com/en-us/devops/deliver/what-is-continuous-delivery)
+
+Blue Green deployment has become a de-facto standard pattern for the release management and operation at infra and application level in kubernetes environments.
+
+In this article is described the design and implementation of the blue green deployment for AKS leveraging Azure Cloud managed services and native kubernetes features. Adoption of this pattern improves the operation and availability during the deployment of changes/upgrades of AKS clusters.
+
+The main benefits of the solution are:
+- Minimized downtime during the release
+- Rollback strategy out of the box
+- Improved control and operation during the release and deployment of AKS changes and upgrades
+- Test for DR procedure
+
+The Azure services that are part of the pattern are listed in the [components](#components) section; below are the main ones:
+- AKS
+- Azure Application Gateway
+- Azure Private DNS
+
+From an automation and CI/CD perspective the solution can be implemented in multiple ways. We suggest:
+- Bicep or Terraform for the IaC
+- Azure Pipelines or Github Actions for the CI/CD
+
+
+## Potential use cases
+This solution is a generalized architecture pattern, which can be used for many different scenarios and industries. In particular it can be applied in AKS deployments and is also used for [Mission Critical scenario](https://docs.microsoft.com/en-us/azure/architecture/reference-architectures/containers/aks-multi-region/aks-multi-cluster). See the following example solutions that build off of this core architecture:
+
+- [Link to first solution idea or other architecture that builds off this solution](filepath.yml)
+
+
+## Architecture
+Below is the high level architecture that describes the pattern and related services invovled. The [Worklfow section](#workflow) describ in detail the steps for the implementation of the pattern and in particular the sequence of events to have the proper switch between the clusters.
```

Comment:
Change "describ" to "describes".

Also rephrase "the sequence of events to have the proper switch between the clusters." as it is not clear what you mean.

Reply:
rephrased in  "including the sequence of events to have the proper traffic switch, from networking and app perspcetve, between the blue and green clusters."